src/matrix/kernelmatrix.jl

```diff
@@ -103,6 +103,14 @@ function kernelmatrix(κ::SimpleKernel, x::AbstractVector, y::AbstractVector)
     return map(d -> kappa(κ, d), pairwise(metric(κ), x, y))
 end
 
+function kerneldiagmatrix(κ::SimpleKernel, x::AbstractVector)
+    return map(x -> κ(x, x), x)
+end
+
+function kerneldiagmatrix(κ::SimpleKernel, x::AbstractVector, y::AbstractVector)
+    return map(d -> kappa(κ, d), map(metric(κ), x, y))
```

Comment:
```suggestion
    return map(d -> kappa(κ, d), colwise(metric(κ), x, y))
```
The solution isn't currently working because `map` utilises `getindex` on `x` and `y` to compute the distances. You would need to use an implementation that doesn't do this -- `Distances.colwise` would satisfy this.

Reply:
Oh I see, where `Distances.colwise` is not defined for vectors right now, but we could have our own implementation.

Reply:
Yeah, we'll probably want to do what we did with `pairwise` and roll our own to avoid type-pirating as you say.

Reply:
This still does not work, or maybe my implementation of `colwise` is not what you thought about